Retarus/retarus_modules/retarus_events_comsumer.py

```diff
@@ -0,0 +1,62 @@
+from queue import SimpleQueue
+from threading import Event, Thread
+
+import orjson
+from sekoia_automation.connector import Connector
+from websocket import WebSocketApp, WebSocketTimeoutException
+
+
+class RetarusEventsConsumer(Thread):
+    def __init__(self, connector: Connector, queue: SimpleQueue):
+        super().__init__()
+        self.connector = connector
+        self.queue = queue
+        self.configuration = connector.configuration
+        self.websocket: WebSocketApp | None = None
+        self._stop_event = Event()
+
+    def create_websocket(self):
+        return WebSocketApp(
+            url=self.configuration.ws_url,
+            header=[f"Authorization: Bearer {self.configuration.ws_key}"],
+            on_message=self.on_message,
+            on_error=self.on_error,
+            on_close=self.on_close,
+        )
+
+    def on_message(self, _, event) -> None:
+        try:
+            message = orjson.loads(event)
+            self.queue.put(message)
+        except orjson.JSONDecodeError as ex:
+            self.connector.log_exception(ex, message="Failed to consume event")
+
+    def on_error(self, _, error):
+        if isinstance(error, WebSocketTimeoutException):
+            # add a gentler message for timeout
+            self.connector.log(message="Websocket timed out", level="warning")
+        else:
+            self.connector.log(message=f"Websocket error: {error}", level="error")
+
+    def on_close(self, *_):
+        self.connector.log(message="Closing socket connection", level="info")
+
+    @property
+    def is_running(self):
+        return not self._stop_event.is_set()
+
+    def run(self):  # pragma: nocover
+        self.connector.log(message=f"Connection to stream {self.configuration.ws_url}", level="info")
+
+        while self.is_running:
+            self.websocket = self.create_websocket()
+            teardown = self.websocket.run_forever()
```

Comment:
Not sure to understand how the websocket connection will be closed when `self.is_running` is `False`.

I think that `self.websocket.run_forever()` is supposed to be blocking indefinitely.

Reply:
`run_forever` is supposed to return if the websocket is closed

I took this code from another module, the way I understand it `self.is_running` won't stop the connection by itself, it is just here to prevent the worker to reopen the connection once it's closed. But it requires the connection to be closed by other means 

Now that you mention it though it's a workflow that I could improve a little, and try to make it so that when `not self.is_running` the websocket is closed